iml-manager-cli/src/main.rs

```diff
@@ -18,9 +18,101 @@ pub enum StratagemCommand {
         /// The name of the filesystem to scan
         #[structopt(short = "fs", long = "filesystem")]
         fs: String,
+        /// The report duration
+        #[structopt(short = "rd", long = "report", parse(try_from_str = "parse_duration"))]
+        rd: Option<u32>,
+        /// The purge duration
+        #[structopt(short = "pd", long = "purge", parse(try_from_str = "parse_duration"))]
+        pd: Option<u32>,
     },
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+struct StratagemCommandData {
+    filesystem: String,
+    report_duration: Option<u32>,
+    purge_duration: Option<u32>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum DurationErrorKind {
+    /// No unit provided
+    NoUnit,
+    /// Invalid Unit provided
+    InvalidUnit,
+    /// Invalid value
+    InvalidValue,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct DurationParseError {
+    kind: DurationErrorKind,
+}
+
+impl DurationParseError {
+    pub fn to_string(&self) -> String {
+        match self.kind {
+            DurationErrorKind::NoUnit => String::from("No unit was specified."),
+            DurationErrorKind::InvalidUnit => {
+                String::from("Invalid unit provided. Valid units include: h, d")
+            }
+            DurationErrorKind::InvalidValue => String::from(
+                "Invalid value provided. Duration must be in the form of '<u32 value>{h|d}'",
+            ),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum DurationUnit {
+    /// Duration in hours
+    Hours,
+    /// Duration in days
+    Days,
+}
+
+fn parse_duration(src: &str) -> Result<u32, DurationParseError> {
+    if src.len() < 2 {
+        return Err(DurationParseError {
+            kind: DurationErrorKind::InvalidValue,
+        });
+    }
+
+    let val = &src[0..src.len() - 1];
+    let unit = &src[src.len() - 1..];
```

Comment:
```rust
let mut val = String::from(src);
let unit = val.pop();
```

Reply:
Updated. Is there an advantage to doing it this way as opposed to a slice?

Reply:
The advantage is `unit` is a `char` instead of `&str` so `unit.chars().last()` below is not needed.

Reply:
Another advantage is there is not possibility of panic since were not indexing.